src/hmem_cuda.c

```diff
@@ -52,13 +53,27 @@
  * Convenience higher-order macros for enumerating CUDA driver/runtime and
  * NVML API function names
  */
+
+#if HAVE_CUDA_DMABUF
+#define CUDA_DRIVER_FUNCS_DEF(_)	\
+	_(cuGetErrorName)		\
+	_(cuGetErrorString)		\
+	_(cuPointerGetAttribute)	\
+	_(cuPointerSetAttribute)	\
+	_(cuDeviceCanAccessPeer)	\
+	_(cuMemGetAddressRange)		\
+	_(cuMemGetHandleForAddressRange) \
+	_(cuDeviceGetAttribute)		\
+	_(cuDeviceGet)
+#else
 #define CUDA_DRIVER_FUNCS_DEF(_)	\
 	_(cuGetErrorName)		\
 	_(cuGetErrorString)		\
 	_(cuPointerGetAttribute)	\
 	_(cuPointerSetAttribute)	\
 	_(cuDeviceCanAccessPeer)	\
 	_(cuMemGetAddressRange)
+#endif /* HAVE_CUDA_DMABUF */
```

Comment:
It's simpler to conditionally define a dedicated macro for the dmabuf-specific functions and have `CUDA_DRIVER_FUNCS_DEF(_)` unconditionally inherit them. Otherwise, you need to maintain the function list for both conditions, which defeats the purpose of the macros.

```c
#if HAVE_CUDA_DMABUF
#define CUDA_DRIVER_DMABUF_FUNCS_DEF(_) \
  _(cuMemGetHandleForAddressRange)      \
  _(cuDeviceGetAttribute)               \
  _(cuDeviceGet)
#else
#define CUDA_DRIVER_DMABUF_FUNCS_DEF(_)
#endif

#define CUDA_DRIVER_FUNCS_DEF(_)  \
  _(cuGetErrorName)               \
  /* ... */                       \
  CUDA_DRIVER_DMABUF_FUNCS_DEF(_)

Reply:
Thanks. Updated

Reply:
I moved `cuDeviceGetAttribute` and `cuDeviceGet` from `CUDA_DRIVER_DMABUF_FUNCS_DEF` because they are not necessarily used for dmabuf usage and should be generally available for older cuda versions